db-engine/src/main/java/com/splicemachine/db/iapi/types/UserType.java

```diff
@@ -257,21 +253,10 @@ public Object getObject()
 
 	@Override
 	public Object getSparkObject() throws StandardException {
-		ByteArrayOutputStream bos = new ByteArrayOutputStream();
-		ObjectOutput out = null;
-		try {
-			out = new ObjectOutputStream(bos);
-			out.writeObject(value);
-			out.flush();
-			return bos.toByteArray();
-		} catch (IOException e) {
-			throw new RuntimeException(e);
-		} finally {
-			try {
-				bos.close();
-			} catch (IOException ex) {
-				// ignore close exception
-			}
+		if (value instanceof Serializable) {
+			return SerializationUtils.serialize((Serializable) value);
+		} else {
+			return super.getSparkObject();
```

Comment:
How about this?
```java
return (value instanceof Serializable)
	? SerializationUtils.serialize((Serializable) value)
	: super.getSparkObject();
```

Reply:
I think that looks good, but it's a matter of personal taste.